python/packages/ai/teams/ai/augmentations/tools_augmentation.py

```diff
@@ -0,0 +1,189 @@
+"""
+Copyright (c) Microsoft Corporation. All rights reserved.
+Licensed under the MIT License.
+"""
+
+from __future__ import annotations
+
+import json
+from typing import List, Optional, Union, cast
+
+from botbuilder.core import TurnContext
+
+from ...state import MemoryBase
+from ..models.chat_completion_action import ChatCompletionAction
+from ..models.prompt_response import PromptResponse
+from ..planners.plan import (
+    Plan,
+    PredictedCommand,
+    PredictedDoCommand,
+    PredictedSayCommand,
+)
+from ..prompts.message import ActionCall, Message
+from ..prompts.sections.prompt_section import PromptSection
+from ..tokenizers.tokenizer import Tokenizer
+from ..validators.validation import Validation
+from .augmentation import Augmentation
+from .tools_constants import ACTIONS_HISTORY, TOOL_CHOICE
+
+
+class ToolsAugmentation(Augmentation[Union[str, List[ActionCall]]]):
+    """
+    A server-side 'tools' augmentation.
+    """
+
+    _actions: List[ChatCompletionAction] = []
+
+    def __init__(self, actions: List[ChatCompletionAction] = []) -> None:
+        # pylint: disable=dangerous-default-value
```

Comment:
its best to either disable the pylint rule if it is not helpful, or find a way to satisfy it with your code. Inline disables are almost never a good idea